google_nest_sdm/webrtc_util.py

```diff
@@ -0,0 +1,114 @@
+# SDP Direction Constants
+DIRECTION_SENDRECV = 'sendrecv'
+DIRECTION_SENDONLY = 'sendonly'
+DIRECTION_RECVONLY = 'recvonly'
+DIRECTION_INACTIVE = 'inactive'
+
+AVAILABLE_DIRECTIONS = [DIRECTION_SENDRECV,
+                        DIRECTION_SENDONLY,
+                        DIRECTION_RECVONLY,
+                        DIRECTION_INACTIVE]
+
+
+class WebRTCUtility:
+    """
+    This class provides methods for managing WebRTC SDP.
+    """
+
+    def get_media_direction(self, sdp, kind):
+        """
+        Retrieves the direction of media tracks from the SDP based on the kind (audio/video).
+
+        Args:
+            sdp (str): The SDP content
+            kind (str): The kind of media track to check ('audio' or 'video').
+
+        Returns:
+            str: The direction of the media track. One of 'sendrecv', 'sendonly', 'recvonly', or 'inactive'.
+        """
+        # Split the SDP into lines
+        lines = sdp.splitlines()
+
+        # Variable to track if we are in the desired media section
+        in_media_section = False
+
+        # Search for the media type and direction
+        for line in lines:
+            # Check if the line is a media description line (m=)
+            if line.startswith('m='):
+                in_media_section = (kind in line)
```

Comment:
Are there multiple subtypes of audio/video? Does it work to check explicitly here `in_media_section = line.startswith(f'm={kind}'):`?

Reply:
yes, it works as well